walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -11,13 +11,130 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
+ 
 package com.google.sps;
-
+ 
+import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.Collection;
-
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+ 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // create a full day of free timeslots
+    List<TimeRange> fullDay = new ArrayList<TimeRange>();
+    fullDay.add(TimeRange.WHOLE_DAY);
+    
+    // cannot modify arraylist in an enhanced for loop, must create separate arraylists
+    List<TimeRange> freeTimes = new ArrayList<TimeRange>();
+    List<TimeRange> blockedTimes = new ArrayList<TimeRange>();
+
+
+    Collection<String> attendees = request.getAttendees();
+    Set<String> eventAttendees;
+    
+  for(Event event: events){
+    eventAttendees = event.getAttendees();
+    boolean isOverlappedAttendee = false;
+    for(String attendee:eventAttendees){
+      if(attendees.contains(attendee)){
+        isOverlappedAttendee = true;
+        // add a break to terminate loop early if at least one attendee is present at event
+        break; 
+        }
+      }
+      if(isOverlappedAttendee){
+        TimeRange blocked = event.getWhen();
+        for(TimeRange timeslot: fullDay){
+          if(blocked.overlaps(timeslot)){
+            // case one: event is fully enclosed within a free timeslot
+            if(blocked.start() > timeslot.start() && blocked.end() < timeslot.end()){
+              freeTimes.add(TimeRange.fromStartEnd(timeslot.start(), blocked.start(),false));
+              freeTimes.add(TimeRange.fromStartEnd(blocked.end(),timeslot.end(),false));
+            }
+            // case two: event overlaps end of free timeslot
+            else if(blocked.start() >= timeslot.start() && blocked.end() >= timeslot.end()){
```

Comment:
Given the above conditional, isn't this equivalent to `else if(blocked.start() == timeslot.start() && blocked.end() == timeslot.end()){` ? If so, it would make it more clear to explicitly check for equality.

Reply:
I moved the explicit test of equality to the first case, where it makes the most sense.